File: 구름 알고리즘 미션/그래프 이론/폭탄 해제/폭탄 해제.py
import sys
input = sys.stdin.readline

N, M = map(int, input().split())
connectLine = [[]] * M
connectCnt = [0] * N

for i in range(M):
	A, B = map(int, input().split())
	connectLine[i] = [A, B]
	connectCnt[A-1] += 1
	connectCnt[B-1] += 1
	
result = []
for i in range(M):
	A, B = connectLine[i]
	if connectCnt[A-1] > 1 and connectCnt[B-1] > 1:
		result.append(i+1)

if(len(result)):
	print(*result)
else:
	print(-1)

File: 구름 알고리즘 미션/그래프 이론/폭탄 해제/폭탄 해제.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const [N, M] = input[0].split(" ").map(Number);
const connectLine = [];
const connectCnt = Array(N).fill(0);

for(let i = 1 ; i <= M ; i++){
	const [A, B] = input[i].split(" ").map(Number);
	connectLine.push([A-1, B-1]);
	connectCnt[A-1]++; 
	connectCnt[B-1]++;
}

let result = [];
for(let i = 0 ; i < M ; i++){
	const [A, B] = connectLine[i];
	if(connectCnt[A] > 1 && connectCnt[B] > 1) {
		result.push(i+1);
	}
}

console.log(result.length > 0 ? result.join(" ") : -1);